app/models/ont/run_factory.rb

```diff
@@ -19,46 +21,33 @@ def save(**options)
       return false unless options[:validate] == false || valid?
 
       run.save(validate: false)
-      flowcells.each { |flowcell| flowcell.save(validate: false) }
+      old_flowcells.each(&:destroy)
       true
     end
 
     private
 
-    def flowcells
-      @flowcells ||= []
-    end
+    attr_reader :old_flowcells
+
+    def build_flowcells(flowcell_specs)
+      run.flowcells.clear
 
-    def build_run(attributes)
-      @run = Ont::Run.new
-      attributes.each do |flowcell_spec|
-        # the flowcell requires a library, so if a library does not exist
-        # the flowcell, and therefore factory, will be invalid
+      flowcell_specs.each do |flowcell_spec|
+        # The flowcell requires a library, so if a library does not exist
+        # the flowcell, and therefore factory, will be invalid.
         library = Ont::Library.find_by(name: flowcell_spec[:library_name])
-        flowcells << Ont::Flowcell.new(position: flowcell_spec[:position],
-                                       run: run, library:
-        library)
+        run.flowcells.build(position: flowcell_spec[:position], run: run, library: library)
       end
     end
 
     def check_run
       errors.add('run', 'was not created') if run.nil?
 
-      return if @run.valid?
+      return if run.valid?
 
       run.errors.each do |k, v|
         errors.add(k, v)
       end
     end
-
-    def check_flowcells
```

Comment:
Again I don't know why this validation is removed, unless validating the run validated the flowcells. If so could it be commented somewhere? And a test added to the run_spec.rb?

Reply:
There are tests already that prove this happens.  I don't feel it's justified to add a comment because we don't keep an array in the factory of these objects so it doesn't even feel like they should be validated.  They're stored directly on the run instead so we should expect the run to validate them when we validate it.

Reply:
Since the run validates the flowcells, do we need the run factory tests that check the validation on the flowcell? I guess we do since we wan't to ensure it isn't over-validated

Reply:
Resolving this in favour of the above conversation about the same thing